ports/stm32f4/boards.c

```diff
@@ -156,6 +163,37 @@ void board_app_jump(void)
 {
   volatile uint32_t const * app_vector = (volatile uint32_t const*) BOARD_FLASH_APP_START;
 
+#ifdef BUTTON_PIN
+  HAL_GPIO_DeInit(BUTTON_PORT, BUTTON_PIN);
+#endif
+
+#ifdef LED_PIN
+  HAL_GPIO_DeInit(LED_PORT, LED_PIN);
+#endif
+
+#if NEOPIXEL_NUMBER
+  HAL_GPIO_DeInit(NEOPIXEL_PORT, NEOPIXEL_PIN);
+#endif
+
+#if defined(UART_DEV) && CFG_TUSB_DEBUG
+  HAL_UART_DeInit(&UartHandle);
+  HAL_GPIO_DeInit(UART_GPIO_PORT, UART_TX_PIN | UART_RX_PIN);
+  UART_CLOCK_DISABLE();
+#endif
+
+  __HAL_RCC_GPIOA_CLK_DISABLE();
+  __HAL_RCC_GPIOB_CLK_DISABLE();
+  __HAL_RCC_GPIOC_CLK_DISABLE();
+  __HAL_RCC_GPIOD_CLK_DISABLE();
+
+  HAL_RCC_DeInit();
+
+  __disable_irq();
```

Comment:
it is indeed important to disable irq here, though I could find `__eanble_irq()` is called anywhere in the startup asm (startup_stm32f401xc) Reset_Handler. Would you mind helping to explain when irq got enable, I may miss something.

Reply:
I actually can test w/ this reverted. I had added this when trying to diagnose an issue, which ended up being tracked down to another problem. This may not actually be needed.

Reply:
Worked fine without that. I've removed and fixed up that commit and force pushed.